huaweicloud/services/cdm/resource_huaweicloud_cdm_link.go

```diff
@@ -0,0 +1,310 @@
+	}
+
+	clusterId, linkName := ParseLinkInfoFromId(d.Id())
+
+	resp, gErr := link.Get(client, clusterId, linkName)
+	if gErr != nil {
+		return common.CheckDeletedDiag(d, err, "Error retrieving CDM link")
+	}
+
+	detail := resp.Links[0]
+	mErr := multierror.Append(
+		d.Set("region", region),
+		d.Set("name", detail.Name),
+		d.Set("cluster_id", clusterId),
+		d.Set("connector", detail.ConnectorName),
+		d.Set("enabled", detail.Enabled),
+		setConfigToState(d, detail.LinkConfigValues.Configs),
+	)
+
+	if mErr.ErrorOrNil() != nil {
+		return fmtp.DiagErrorf("Error setting CDM link fields: %s", mErr)
+	}
+
+	return nil
+}
+
+func setConfigToState(d *schema.ResourceData, configs []link.Configs) error {
+	if len(configs) == 0 {
+		return nil
+	}
+
+	result := make(map[string]string)
+	for _, item := range configs {
+		if item.Name == "linkConfig" {
+			for _, v := range item.Inputs {
+				if v.Value != "" {
+					key := strings.Replace(v.Name, configPref, "", 1)
+
+					if key == "password" {
+						d.Set("password", v.Value)
+					} else if key == "securityKey" || key == "sk" {
+						d.Set("security_key", v.Value)
+					} else if key == "accessKey" || key == "ak" {
+						d.Set("access_key", v.Value)
+					} else {
+						result[key] = v.Value
+					}
+				}
+			}
+		}
+	}
+	return d.Set("config", result)
+}
+
+func resourceCdmLinkUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.CdmV11Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CDM v1 client, err=%s", err)
+	}
+
+	clusterId, linkName := ParseLinkInfoFromId(d.Id())
+	newName := d.Get("name").(string)
+
+	opts := link.LinkCreateOpts{
+		Links: []link.Link{
+			{
+				Name:             newName,
+				ConnectorName:    d.Get("connector").(string),
+				Enabled:          utils.Bool(d.Get("enabled").(bool)),
+				LinkConfigValues: buildLinkConfigParamter(d),
+			},
+		},
+	}
+
+	logp.Printf("[DEBUG] Update CDM link opts: %#v", opts)
+
+	_, updateErr := link.Update(client, clusterId, linkName, opts)
+	if updateErr != nil {
+		return fmtp.DiagErrorf("Error update CDM link: %s", updateErr)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", clusterId, newName))
+	return resourceCdmLinkRead(ctx, d, meta)
+}
+
+func resourceCdmLinkDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.CdmV11Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CDM v1 client, err=%s", err)
+	}
+
+	clusterId, linkName := ParseLinkInfoFromId(d.Id())
+
+	_, dErr := link.Delete(client, clusterId, linkName)
+	if dErr != nil {
+		return fmtp.DiagErrorf("delete CDM link failed. %q:%s", d.Id(), dErr)
+	}
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func ParseLinkInfoFromId(id string) (clusterId, linkName string) {
```

Comment:
it seems that this function will not be used outside the package,
it is recommended to use `parseLinkInfoFromId`